src/dataservices/notes.ts

```diff
@@ -1,41 +1,33 @@
-import { DynamoDBClient, DynamoDBClientConfig } from '@aws-sdk/client-dynamodb';
+import { DynamoDBClient } from '@aws-sdk/client-dynamodb';
 import {
   DynamoDBDocumentClient,
   GetCommand,
   GetCommandOutput,
+  BatchGetCommand,
+  BatchGetCommandInput,
+  BatchGetCommandOutput,
 } from '@aws-sdk/lib-dynamodb';
 import config from '../config';
 import { HighlightNote, HighlightNoteEntity } from '../types';
+import { backoff } from './utils';
 
 export class NotesDataService {
-  private client: DynamoDBClient;
   // Easier to work with Document client since it abstracts the types
   public dynamo: DynamoDBDocumentClient;
   private table = config.dynamoDb.notesTable;
 
-  constructor() {
-    const dynamoClientConfig: DynamoDBClientConfig = {
-      region: config.aws.region,
-    };
-    // Set endpoint for local client, otherwise provider default
-    if (config.aws.endpoint != null) {
-      dynamoClientConfig['endpoint'] = config.aws.endpoint;
-    }
-    this.client = new DynamoDBClient(dynamoClientConfig);
+  constructor(private client: DynamoDBClient) {
     this.dynamo = DynamoDBDocumentClient.from(this.client, {
-      marshallOptions: { convertEmptyValues: true },
+      marshallOptions: {
+        convertEmptyValues: true,
+        removeUndefinedValues: true,
```

Comment:
:lolsob: i remember hitting this

Reply:
your comment in prospect-api saved me from having to get burned